### PR TITLE
Fix sar rescaling bug for orfeo

### PR DESCRIPTION
**Pull request by kvantricht**

### CODE REVIEW

**Comment by kvantricht** [comment]
@VincentVerelst @HansVRP one problem here is that with the proposed simplification, a backend check is actually no longer needed. However, changing the function signature has quite some consequences in other libraries making use of this functionality. Not sure how you normally treat this. For now I kept the signature as is.

**Comment by kvantricht** [comment]
FYI got a failed jenkins build for this branch, though with some 500 error I don't understand.

**Comment by HansVRP** [comment]
Hi Kristof, thank you for the contribution. 

We'll take a look and investigate how we can fix the tests and include the backend simplification

File: src/openeo_gfmap/preprocessing/sar.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
...
 
 import openeo
-from openeo.processes import array_create, if_, is_nodata, power
+from openeo.processes import array_create, power
 
-from openeo_gfmap import Backend, BackendContext
+from openeo_gfmap import BackendContext
 
 
 def compress_backscatter_uint16(
```
Comment (VincentVerelst) [inline src/openeo_gfmap/preprocessing/sar.py:9]: I propose we make `backend_context` optional and set it to None by default.

We then add a warning for users who do set it:
```python
import warnings
if backend_context is not None:
    warnings.warn(
                "The `backend_context` argument is deprecated.", DeprecationWarning, stacklevel=2
            )
```

Comment (kvantricht) [reply src/openeo_gfmap/preprocessing/sar.py:9]: that looks good. Can you push that?

Comment (VincentVerelst) [reply src/openeo_gfmap/preprocessing/sar.py:9]: In a meeting now, so go ahead

Comment (kvantricht) [reply src/openeo_gfmap/preprocessing/sar.py:9]: actually i don't know if that will work. Other repos don't use this as kwarg, and rely on the order of two args: `backend_context: BackendContext, cube: openeo.DataCube`.

We cannot simply make the first a kwarg without touching the second.

Comment (HansVRP) [reply src/openeo_gfmap/preprocessing/sar.py:9]: I believe we should prioritize having a correct backscatter correction here, and not push any of the foreseen updates into this PR.

**Review by VincentVerelst — APPROVED** [review]

**Pull request merged**